cmake/DD4hepBuild.cmake

```diff
@@ -565,6 +570,10 @@ function(dd4hep_add_dictionary dictionary )
     LIST(APPEND comp_defs ${def})
   endforeach()
 
+if(DD4HEP_BUILD_DEBUG)
```

Comment:
Apparently yes. ROOT data streaming simply crashed for conditions, where this flag actually changes the structure layout.
Also the TStreamerInfo was wrong.
For this reason the compile flag must be passed to rootcling.

Reply:
The idea is that rootcling is being passed this information from one of the targets (or USE parameter?), if you can tell me more details I will investigate.

Reply:
I do not know. The very best would probably be to generate some "config.h", which contains all relevant compile parameters.
This would ensure that such parameters, which cannot be changed for one release build are also properly propagated
to all client projects....

Reply:
Seems to me this
https://github.com/AIDASoft/DD4hep/blob/c179ef49bb51c3ce537b837b335b7293ab59105b/DDCore/CMakeLists.txt#L43

is missing DDCore and thus not inheriting the DD4HEP_DEBUG.
I did get crashes for various tests without DDCore here when DD4HEP_BUILD_DEBUG was ON

Reply:
And probably also in the other dictionaries in `DDCore/CMakeLists.txt`

Reply:
Yes. These are exactly the dictionaries in question.
But how can DDCore depend on DDCore ?

So your claim is:
If the flags are set in DD4hep explicitly, the statement:

IF(DD4HEP_BUILD_DEBUG MATCHES "ON" OR (CMAKE_BUILD_TYPE MATCHES "DEBUG|Debug" AND NOT DD4HEP_BUILD_DEBUG MATCHES "OFF"))
    target_compile_definitions(DDCore PUBLIC DD4HEP_DEBUG=1)
  ENDIF()

Would do all the job ?
However, it then must move to cmake/DD4hepBuild.cmake so that it is picked up by all clients.

Reply:
The properties are stored in DD4hepConfig-targets.cmake there is no need to move this anywhere else
```cmake
# Create imported target DD4hep::DDCore
add_library(DD4hep::DDCore SHARED IMPORTED)

set_target_properties(DD4hep::DDCore PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "DD4HEP_DEBUG=1"
  INTERFACE_COMPILE_FEATURES "cxx_std_14"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "DD4hep::DDParsers;ROOT::Core;ROOT::Rint;ROOT::Tree;ROOT::Physics;ROOT::Geom;ROOT::GenVector;XercesC::XercesC;dl"
)
```

> But how can DDCore depend on DDCore ?

Thanks to the generator expressions which are evaluated at build time
https://github.com/AIDASoft/DD4hep/blob/c179ef49bb51c3ce537b837b335b7293ab59105b/cmake/DD4hepBuild.cmake#L572